sympy/core/mod.py

```diff
@@ -57,6 +57,8 @@ def doit(p, q):
 
             # by ratio
             r = p/q
+            if r.is_integer:
+                return S.Zero
             try:
                 d = int(r)
```

Comment:
Can the lines below the added code be removed? I can't think of a situation where `int(r)` would succeed but `r.is_integer` is not True.

Reply:
Is this change required?

Reply:
I'm just asking a question since you have changed the code here. Do the lines that you have added replace the lines below?

Reply:
I am not sure. This was a change suggested by you in my previous PR (https://github.com/sympy/sympy/pull/18277#issuecomment-573423619).
You didn't remove the below lines at that time. So, I guess they might not replace these.

Reply:
> I can't think of a situation where int(r) would succeed but r.is_integer is not True.

Even if `r` is not an integer `int(r)` will try to find the floor of the value:
```
In [2]: y = sqrt(5)

In [3]: int(y)
Out[3]: 2
```

So , I guess no changes are required here.